realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1630,9 +1640,21 @@ public void delete(Class<? extends RealmModel> clazz) {
 
 
     @SuppressWarnings("unchecked")
-    private <E extends RealmModel> E copyOrUpdate(E object, boolean update, Map<RealmModel, RealmObjectProxy> cache) {
+    private <E extends RealmModel> E copyOrUpdate(E object, boolean update, Map<RealmModel, RealmObjectProxy> cache, Set<ImportFlag> flags) {
         checkIfValid();
-        return configuration.getSchemaMediator().copyOrUpdate(this, object, update, cache);
+        if (!isInTransaction()) {
+            throw new IllegalStateException("`copyOrUpdate` can only be called inside a write transaction.");
+        }
+        try {
+            return configuration.getSchemaMediator().copyOrUpdate(this, object, update, cache, flags);
+        } catch (IllegalStateException e) {
+            // Convert OS exception to match old API
```

Comment:
We need a TODO referencing an enhancement issue, we can't rely on matching strings for error code behaviour

Reply:
Added a reference to https://github.com/realm/realm-java/issues/6262